metrics-core/src/test/java/software/amazon/swage/metrics/record/cloudwatch/MetricDataAggregatorTest.java

```diff
@@ -135,82 +145,91 @@ public void aggregate() throws Exception {
-                assertEquals("Aggregated metric has wrong stats count", Double.valueOf(1), stats.getSampleCount());
-                seen[1] = true;
+                final StatisticSet exp = toStatisticSet(6.28);
+                assertEquals(exp, stats);
+                seen[2]++;
             }
             else if (d.getMetricName().equals(D.toString()) && d.getUnit().equals(StandardUnit.Milliseconds.toString())) {
-                StatisticSet stats = d.getStatisticValues();
-                assertEquals("Aggregated metric has wrong stats value", Double.valueOf(42+9+4.1), stats.getSum());
-                assertEquals("Aggregated metric has wrong stats value", Double.valueOf(4.1), stats.getMinimum());
-                assertEquals("Aggregated metric has wrong stats value", Double.valueOf(42), stats.getMaximum());
-                assertEquals("Aggregated metric has wrong stats count", Double.valueOf(3), stats.getSampleCount());
-                seen[2] = true;
+                final StatisticSet exp0 = toStatisticSet(42.0);
+                final StatisticSet exp1 = toStatisticSet(9.0, 2);
+
+                if (exp0.equals(stats)) {
+                    seen[3]++;
+                } else if (exp1.equals(stats)) {
+                    seen[4]++;
+                } else {
+                    fail(String.format("Aggregated metric was [%s], expected [%s] or [%s]",
+                            stats, exp0, exp1));
+                }
             }
             else if (d.getMetricName().equals(D.toString()) && d.getUnit().equals(StandardUnit.Percent.toString())) {
-                StatisticSet stats = d.getStatisticValues();
-                assertEquals("Aggregated metric has wrong stats value", Double.valueOf(2), stats.getSum());
-                assertEquals("Aggregated metric has wrong stats value", Double.valueOf(2), stats.getMinimum());
-                assertEquals("Aggregated metric has wrong stats value", Double.valueOf(2), stats.getMaximum());
-                assertEquals("Aggregated metric has wrong stats count", Double.valueOf(1), stats.getSampleCount());
-                seen[3] = true;
+                final StatisticSet exp = toStatisticSet(2.0);
+                assertEquals(exp, stats);
+                seen[5]++;
             }
             else {
                 fail("Unexpected metric returned in aggregated set");
             }
         }
         for (int i=0; i<seen.length; i++) {
-            assertTrue("Expected aggregated metric not seen", seen[i]);
+            assertEquals("Expected aggregated metric not seen exactly once", 1, seen[i]);
         }
 
         assertTrue("Flush with no attributes was non-empty", aggregator.flush().isEmpty());
 
         // Now add more attributes, but let's just do two this time
         for (int i=0; i<5; i++) {
-            aggregator.add(context, A, i, StandardUnit.Count);
+            aggregator.add(context, A, 1, StandardUnit.Count);
```

Comment:
Note that the behavior of this test has changed. It used to publish [0, 1, 2, ...] or [0, .01, .02, ...] and expect them to be aggregated. With lossless metrics, they will not. Since the goal of the test seemed to be aggregation, the test was changed so aggregation still occurs.